src/module.c

```diff
@@ -7767,9 +7767,9 @@ void moduleBlockedClientTimedOut(client *c) {
     long long prev_error_replies = server.stat_total_error_replies;
     bc->timeout_callback(&ctx,(void**)c->argv,c->argc);
     moduleFreeContext(&ctx);
-    if (!bc->blocked_on_keys) {
-        updateStatsOnUnblock(c, bc->background_duration, 0, server.stat_total_error_replies != prev_error_replies);
-    }
+    long blocked_duration = bc->blocked_on_keys ? 0 : bc->background_duration;
```

Comment:
do we really need to take extra care not to use background_duration on blocked keys? it'll be 0 anyway. am i missing something?

Reply:
to tell the truth, IDK. I only tried to keep this aligned with existing logic. @guybe7 do you know if we always take the background duration? (I think it is only updated by specific call to RM_BlockedClientMeasureTimeEnd)

Reply:
Guy is currently away.
what i meant was that since it's only updated in RM_BlockedClientMeasureTimeEnd, it should be otherwise set to 0, and then we can avoid the condition here and always sum it.
i.e. even if we call RM_BlockClientOnKeys we set it to 0..

Reply:
O.K - I basically agree. will do